bin/helpers.sh

```diff
@@ -0,0 +1,267 @@
+		echo "WP-CLI is not installed. Exiting."
+		exit 1
+	fi
+
+	echo "Downloading WordPress version: ${WP_VERSION} to ${TMPDIR}/wordpress"
+	wp core download --version="$WP_VERSION" --path="${TMPDIR}/wordpress" --force
+}
+
+# Sets up WordPress using wp config create (if a wp-config file doesn't already exist) and wp core install. Expects that WordPress is already downloaded.
+setup_wp() {
+	# Initialize variables with default values
+	TMPDIR="/tmp"
+	DB_NAME="wordpress_test"
+	DB_USER="root"
+	DB_PASS=""
+	DB_HOST="127.0.0.1"
+	WP_VERSION=${WP_VERSION:-latest}
+
+	# Parse command-line arguments
+	for i in "$@"; do
+		case $i in
+			--dbname=*)
+			DB_NAME="${i#*=}"
+			;;
+			--dbuser=*)
+			DB_USER="${i#*=}"
+			;;
+			--dbpass=*)
+			DB_PASS="${i#*=}"
+			;;
+			--dbhost=*)
+			DB_HOST="${i#*=}"
+			;;
+			--version=*)
+			WP_VERSION="${i#*=}"
+			;;
+			--tmpdir=*)
+			TMPDIR="${i#*=}"
+			;;
+			*)
+			# unknown option
+			echo "Unknown option: $i. Usage: setup_wp --dbname=wordpress_test --dbuser=root --dbpass=root --dbhost=localhost --version=latest --tmpdir=/tmp"
+			exit 1
+			;;
+		esac
+	done
+
+	download http://api.wordpress.org/core/version-check/1.7/ "$TMPDIR"/wp-latest.json
+
+	if [ ! -f "$TMPDIR/wordpress/wp-config.php" ]; then
+		echo "Creating wp-config.php"
+		wp config create --dbname="$DB_NAME" --dbuser="$DB_USER" --dbpass="$DB_PASS" --dbhost="$DB_HOST" --dbprefix="wptests_" --path="${TMPDIR}/wordpress"
+	fi
+		
+	wp core install --url=localhost --title=Test --admin_user=admin --admin_password=password --admin_email=test@dev.null --path="${TMPDIR}/wordpress"
+}
+
+# Sets up WordPress nightly version. Uses setup_wp to get WordPress, then installs Gutenberg on the recently installed WordPress site.
+setup_wp_nightly() {
+	# Initialize variables with default values
+	TMPDIR="/tmp"
+	DB_NAME="wordpress_test"
+	DB_USER="root"
+	DB_PASS=""
+	DB_HOST="127.0.0.1"
+
+	# Parse command-line arguments
+	for i in "$@"; do
+		case $i in
+			--dbname=*)
+			DB_NAME="${i#*=}"
+			;;
+			--dbuser=*)
+			DB_USER="${i#*=}"
+			;;
+			--dbpass=*)
+			DB_PASS="${i#*=}"
+			;;
+			--dbhost=*)
+			DB_HOST="${i#*=}"
+			;;
+			--tmpdir=*)
+			TMPDIR="${i#*=}"
+			;;
+			*)
+			# unknown option
+			echo "Unknown option: $i. Usage: setup_wp --dbname=wordpress_test --dbuser=root --dbpass=root --dbhost=localhost --version=latest --tmpdir=/tmp"
+			exit 1
+			;;
+		esac
+	done
+
+	WP_DIR="$TMPDIR/wordpress"
+
+	setup_wp --version="nightly" --tmpdir="$TMPDIR" --dbname="$DB_NAME" --dbuser="$DB_USER" --dbpass="$DB_PASS" --dbhost="$DB_HOST"
+
+	# If nightly version of WP is installed, install latest Gutenberg plugin and activate it.
+	echo "Installing Gutenberg plugin"
+	wp plugin install gutenberg --activate --path="$WP_DIR"
+}
+
+# Gets the WordPress version number from the JSON file. If the version is "latest", it will get the latest version from the JSON file. If the version is "nightly", it will return "trunk".
+get_wp_version_num() {
```

Comment:
Because there are unit tests in pantheon-mu-plugin that explicitly use the json file to determine the version. See https://github.com/pantheon-systems/pantheon-mu-plugin/blob/main/tests/phpunit/test-pantheon-updates.php#L27-L43

the `wp-latest.json` file will report the _newest_ version of WordPress, which is not the same as the _currently installed_ version of WordPress and there are times when we need to know the difference between the two.